_pages/docs/matching-options.md

```diff
@@ -124,6 +128,31 @@ The `reduce` element gives a way to back off the scores in such cases. The
 element under `algorithms`. The `weight` attribute will be subtracted from the
 score if the algorithm matches.
 
+### Actions
+
+You can provide your own custom action handler functions. Custom actions are called after merge and notify have run. These actions will be called in a separate transaction, so the results of each call will be visible to the actions called after it.
+
+```xml
+<actions>
+  <action name="my-custom-action" function="custom-action" namespace="http://marklogic.com/smart-mastering/action" at="/custom-action.xqy" />
+</actions>
+```
+
+Each custom action can define four properties:
+- **name** - the name of the custom action
+- **function** - the name of the custom function to invoke
+- **namespace** - the namespace of the module containing the function
+- **at** - the uri location of the module containing the function
+
+Your custom action will be called with three parameters:
```

Comment:
@dmcassel @paxtonhare I'm late to the game - how about passing the action XML element to the function as well? That allows the client to store e.g. child elements in the action XML for configuring how the action works. The action function can then look for those child elements to customize its behavior.